src/pkg/path/drive.go

```diff
@@ -27,6 +29,11 @@ func ToDrivePath(p Path) (*DrivePath, error) {
 			With("path_folders", p.Folder(false))
 	}
 
+	if p.Service() == GroupsService {
+		// Groups have an extra /sites/<siteID> in the path
+		return &DrivePath{DriveID: folders[3], Root: folders[4], Folders: folders[5:]}, nil
+	}
```

Comment:
Hmm, this brings me back to the concern about one-off service handling, and that we're doing too much of that.  As a preface: this is a more managable concern than other palces, and we're already doing a _lot_ of this, so I won't block the PR on it, but I do want to voice said concern.

Ideally, I think this func should accept path.Elements instead of a path, where that builder only holds the path tail containing the drive ID and onward.  And hat we hold the assumption that the caller knows how to shorten a path to that tail.  That way we can skip all "if service foo then bar" checks in favor of expecting a well-formed list of elements from packages which should already own the specific format of their path.

Reply:
Hmm, I understand your concern. I am not sure if passing in a trimmed path would be easy enough. We could however consider moving this to a service specific path func. In any case let me drop a TODO for now to be picked up later.